nautobot/core/views/generic.py

```diff
@@ -138,6 +139,9 @@ def get(self, request, *args, **kwargs):
 
         plugin_tabs = _get_registered_content(instance, "detail_tabs", temp_fake_context, return_html=False)
 
+        # TODO: this feels inelegant - should the tabs lookup be a dedicated endpoint rather than piggybacking
+        # on the object-retrieve endpoint?
+        # TODO: similar functionality probably needed in NautobotUIViewSet as well, not currently present
```

Comment:
This should be another API endpoint, similar to plugin_full_width_page, not hacked into the UI code.